sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py

```diff
@@ -173,13 +177,9 @@ def _parse_event(event_data_str: str) -> Tuple[str, StreamEventData]:
         AgentStreamEvent.THREAD_MESSAGE_COMPLETED.value,
         AgentStreamEvent.THREAD_MESSAGE_INCOMPLETE.value,
     }:
-        thread_message = cast(ThreadMessageGenerated, _safe_instantiate(ThreadMessageGenerated, parsed_data))
-        event_obj = ThreadMessage(thread_message)
+        event_obj = _safe_instantiate(ThreadMessage, parsed_data, generated_class=ThreadMessageGenerated)
```

Comment:
Why do we filter out the parameters specific fot the child class?

Reply:
My originally attempt is to use the subclass.   But with the subclass, it filters out all properties define in the auto generated ThreadMessage class.   So id, for example is None after I call _safe_instantiate.

Reply:
Actually, I think you might misunderstood.   You properly wanted to ask, "Why filter out the parameters specific for the parent class."   Anyway, My previous message is my answer.